I/O operation on closed file

The latest version has broken our CI, but it might be that you've just exposed an issue in `dask` and I actually need to create a ticket there instead. This test worked fine at `fsspec==2022.7.1`.

## Callstack ##
```
_______________________ TestSortLargeCSV.test_one_column _______________________
farmlib/core/helpers/tests/test_io.py:657: in test_one_column
    sort_large_csv(self.input_filename,
farmlib/core/helpers/io.py:742: in sort_large_csv
    events = dd.read_csv(input_file, blocksize=blocksize)
venv/lib/python3.8/site-packages/dask/dataframe/io/csv.py:744: in read
    return read_pandas(
venv/lib/python3.8/site-packages/dask/dataframe/io/csv.py:548: in read_pandas
    b_out = read_bytes(
venv/lib/python3.8/site-packages/dask/bytes/core.py:[149](https://gitlab.com/jbariskmanagement/code/farmlib/-/jobs/2957863814#L149): in read_bytes
    values = [
venv/lib/python3.8/site-packages/dask/bytes/core.py:[150](https://gitlab.com/jbariskmanagement/code/farmlib/-/jobs/2957863814#L150): in <listcomp>
    delayed_read(
venv/lib/python3.8/site-packages/dask/delayed.py:695: in __call__
    return call_function(
venv/lib/python3.8/site-packages/dask/delayed.py:662: in call_function
    args2, collections = unzip(map(unpack_collections, args), 2)
venv/lib/python3.8/site-packages/dask/delayed.py:38: in unzip
    out = list(zip(*ls))
venv/lib/python3.8/site-packages/dask/delayed.py:93: in unpack_collections
    if is_dask_collection(expr):
venv/lib/python3.8/site-packages/dask/base.py:187: in is_dask_collection
    return x.__dask_graph__() is not None
venv/lib/python3.8/site-packages/fsspec/core.py:212: in __getattr__
    return getattr(self.f, item)
venv/lib/python3.8/site-packages/fsspec/core.py:149: in f
    raise ValueError(
E   ValueError: I/O operation on closed file. Please call open() or use a with context
```

## Repro case ##
```
class TestSortLargeCSV:
    @pytest.fixture(autouse=True)
    def setup_method(self):
        self.temp_dir = TemporaryDirectory()
        self.input_filename = os.path.join(self.temp_dir.name, "input.csv")

    def test_one_column(self):
        df = pd.DataFrame(columns=["col1"],
                          data=[["a"], ["b"], ["z"], ["x"]])
        df.to_csv(self.input_filename, index=False)

        sort_large_csv(self.input_filename,
                       self.output_filename,
                       index_column="col1",
                       blocksize=self.blocksize)

```

--- a/fsspec/core.py
+++ b/fsspec/core.py
@@ -121,10 +121,6 @@ class OpenFile:
     def __exit__(self, *args):
         self.close()
 
-    def __del__(self):
-        if hasattr(self, "fobjects"):
-            self.close()
-
     @property
     def full_name(self):
         return _unstrip_protocol(self.path, self.fs)

--- a/fsspec/implementations/tests/test_arrow.py
+++ b/fsspec/implementations/tests/test_arrow.py
@@ -108,7 +108,7 @@ def test_rm(fs, remote_dir):
     fs.touch(remote_dir + "/dir/a")
     fs.touch(remote_dir + "/dir/b")
     fs.mkdir(remote_dir + "/dir/c/")
-    fs.touch(remote_dir + "/dir/c/a/")
+    fs.touch(remote_dir + "/dir/c/a")
     fs.rm(remote_dir + "/dir", recursive=True)
     assert not fs.exists(remote_dir + "/dir")
 

